packages/nx-sonarqube/src/executors/scan/utils/utils.ts

```diff
@@ -48,10 +47,30 @@ class EnvMarshaller implements OptionMarshaller {
   }
 }
 
+function getExecutor({ executor }: ScanExecutorSchema): Executor {
+  if (executor === 'vite') {
+    return 'vite';
+  }
+
+  return 'jest';
+}
+
+type CoverageDirectoryName = 'coverageDirectory' | 'reportsDirectory';
+
+function getCoverageDirectoryName(executor: Executor): CoverageDirectoryName {
+  if (executor === 'vite') {
+    return 'reportsDirectory';
+  }
+
+  return 'coverageDirectory';
+}
+
+
 export async function determinePaths(
   options: ScanExecutorSchema,
   context: ExecutorContext
 ): Promise<{ lcovPaths: string; sources: string }> {
+  const executor: Executor = getExecutor(options);
```

Comment:
Bring the executor into the dep level and look at `dep.testTarget.executor` then the function to know which would be:

```ts
type ScanExecutorSchema = '@nx/jest:jest' | '@nx/vite:test';

function getExecutor({ executor }: ScanExecutorSchema): Executor {
  if (executor === '@nx/vite:test') {
    return 'vite';
  }

  return 'jest';
}

```

Reply:
Good idea!